File: Leetcode 217 - contains-duplicate/217.contains-duplicate.cpp
/*
 * @lc app=leetcode id=217 lang=cpp
 *
 * [217] Contains Duplicate
 */

// @lc code=start
#include <algorithm>
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
  bool containsDuplicate(vector<int> &nums) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    unordered_map<int, int> count;
    for (int c : nums) {
      count[c]++;
      if (count[c] >= 2) {
        return true;
      }
    }

    // std::sort(nums.begin(), nums.end());

    // for (int i = 1; i < nums.size(); ++i) {
    //   if (nums[i] == nums[i - 1]) {
    //     return true;
    //   }
    // }
    // return false;

    return false;
  }
};

int main() {
  Solution sol;

  cout << boolalpha;

  vector<int> nums1 = {1, 2, 3, 4}; // false
  cout << sol.containsDuplicate(nums1);
  cout << endl;

  vector<int> nums2 = {1, 2, 3, 1};
  cout << sol.containsDuplicate(nums2); // true
  cout << endl;

  vector<int> nums3 = {1, 1, 1, 3, 3, 4, 3, 2, 4, 2}; // true
  cout << sol.containsDuplicate(nums3);
  cout << endl;
}
// @lc code=end

File: Leetcode 217 - contains-duplicate/217.contains-duplicate.java
/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */

// @lc code=start

import java.util.HashSet;

class Solution {

  public boolean containsDuplicate(int[] nums) {
    HashSet<Integer> seen = new HashSet<>();
    for (int num : nums) {
      if (!seen.add(num)) {
        return true;
      }
    }
    return false;
  }

  public static void main(String[] args) {
    Solution solution = new Solution();
    int[] nums = { 1, 2, 3, 4 };
    System.out.println(solution.containsDuplicate(nums));

    int[] nums2 = { 1, 1, 1, 3, 3, 4, 3, 2, 4, 2 };
    System.out.println(solution.containsDuplicate(nums2));

    int[] nums3 = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
    System.out.println(solution.containsDuplicate(nums3));
  }
}
// @lc code=end
